### ISSUE TITLE
This breaks #eager_load

### ISSUE THREAD

**Issue by kwstannard** [khiav223577/left_joins#6]
`eager_load` by default in Rails 4 uses LEFT OUTER JOINS.

`eager_load` in this gem uses INNER JOINS.

I think that this might be the cause: https://github.com/khiav223577/left_joins/blob/98e36f9643a1484aad0bab82bc987d51797f9008/lib/left_joins.rb#L42

**Comment by khiav223577** [comment]
Thanks for reporting the bug. I'll take it a look later.

Never use `eager_load` before in my company. Do you know what is the difference between `eager_load` and `left_joins`?

**Comment by kwstannard** [comment]
https://blog.bigbinary.com/2013/07/01/preload-vs-eager-load-vs-joins-vs-includes.html

I am not an expert, but it seems like the main difference is that eager_load does other things as well like adding things into the select statement.

**Comment by khiav223577** [comment]
I just released v1.0.4. I believe it will fix the issue :)

### in 1.0.3
```rb
User.eager_load(:posts).count
# SELECT COUNT(*) FROM `users` INNER JOIN `posts` ON `posts`.`user_id` = `users`.`id`
```

### in 1.0.4

```rb
User.eager_load(:posts).count
# SELECT COUNT(DISTINCT `users`.`id`) FROM `users` LEFT OUTER JOIN `posts` ON `posts`.`user_id` = `users`.`id`
```

**Comment by kwstannard** [comment]
I can confirm that it is generating the correct queries as well. Thank you for the fast turn around.
